json_serializable/lib/src/json_key_utils.dart

```diff
@@ -68,7 +68,7 @@ JsonKey _from(FieldElement element, JsonSerializable classAnnotation) {
       // TODO(kevmoo): Support calling function for the default value?
       badType = 'Function';
     } else if (!reader.isLiteral) {
-      badType = dartObject.type.name;
+      badType = dartObject.type.getDisplayString();
```

Comment:
Lets be consistent about `.element.name`.

Reply:
fixed

Reply:
The others too :)